ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/AttestationUtil.java

```diff
@@ -161,43 +196,58 @@ public static UnsignedLong get_current_epoch_boundary_attesting_balance(BeaconSt
+      throws Exception {
     // Get previous epoch
     UnsignedLong previous_epoch = BeaconStateUtil.get_previous_epoch(state);
 
-    // Get previous_epoch_attestations
+    // Get current_epoch_attestations
     List<PendingAttestation> previous_epoch_attestations =
         AttestationUtil.get_epoch_attestations(state, previous_epoch);
 
-    // Get previous_epoch_boundary_attestations
-    List<PendingAttestation> previous_epoch_boundary_attestations =
-        AttestationUtil.get_previous_epoch_boundary_attestations(
-            state, previous_epoch_attestations);
-
-    // Get previous_epoch_boundary_attester_indices
-    List<Integer> previous_epoch_boundary_attester_indices =
-        AttestationUtil.get_attester_indices(state, previous_epoch_boundary_attestations);
+    List<PendingAttestation> previous_epoch_head_attestations = new ArrayList<>();
+    for (PendingAttestation attestation : previous_epoch_attestations) {
+      if (attestation
+          .getData()
+          .getBeacon_block_root()
+          .equals(BeaconStateUtil.get_block_root(state, attestation.getData().getSlot()))) {
+        previous_epoch_head_attestations.add(attestation);
+      }
+    }
+    return previous_epoch_head_attestations;
+  }
 
-    // Get previous_epoch_boundary_attesting_balance
-    UnsignedLong previous_epoch_boundary_attesting_balance =
-        AttestationUtil.get_total_attesting_balance(
-            state, previous_epoch_boundary_attester_indices);
+  public static List<Integer> get_previous_epoch_head_attester_indices(BeaconState state)
+      throws Exception {
+    List<PendingAttestation> previous_epoch_head_attestations =
+        get_previous_epoch_head_attestations(state);
 
-    return previous_epoch_boundary_attesting_balance;
+    return get_attester_indices(state, previous_epoch_head_attestations);
   }
 
-  public static List<Integer> get_previous_epoch_head_attester_indices(BeaconState state) {
-    // todo
-    return new ArrayList<Integer>();
-  }
+  public static UnsignedLong get_previous_epoch_head_attesting_balance(BeaconState state)
+      throws Exception {
+    List<Integer> previous_epoch_head_attester_indices =
+        get_previous_epoch_head_attester_indices(state);
 
-  public static UnsignedLong get_previous_epoch_head_attesting_balance(BeaconState state) {
-    // todo
-    return UnsignedLong.ZERO;
+    return AttestationUtil.get_total_attesting_balance(state, previous_epoch_head_attester_indices);
   }
 
-  public static List<Integer> get_previous_epoch_attester_indices(BeaconState state) {
-    // todo
-    return new ArrayList<Integer>();
+  public static List<Integer> get_previous_epoch_attester_indices(BeaconState state)
+      throws Exception {
+    UnsignedLong previous_epoch = BeaconStateUtil.get_previous_epoch(state);
+
+    List<PendingAttestation> previous_epoch_attestations =
+        AttestationUtil.get_epoch_attestations(state, previous_epoch);
```

Comment:
You don't need to prefix the call to `get_epoch_attestations ` with `AttestationUtil` since this code is in `AttestationUtil`.